### PR TITLE
Feat/typings

### LINKED ISSUE #3: RPC method handler should include parameter types in the function definition.
typedoc to do language documentation

### PR DESCRIPTION
**Pull request by BelfordZ**
fixes #3 #27 #28

### CODE REVIEW

File: templates/js/templated/exported-class.template.js [hunk 1]
```diff
@@ -14,21 +19,49 @@ export default class <%= className %> {
...
     });
   }
 
   <% methods.forEach((method, i) => { %>
-  <%= method.name %>(...params: Array<any>): jayson.JSONRPCRequest {
-    params.forEach((param, i) => this.validators['<%= method.name %>'][i].validate(param));
+  /**
+   * <%= method.summary %>
+   */
+  <%= method.name %>(<%= _.map(method.params, (param) => param.name + ': ' + typeDefs[makeIdForMethodContentDescriptors(method, param)].typeName).join(', ') %>): jayson.JSONRPCRequest {
```
Comment (shanejonas) [inline templates/js/templated/exported-class.template.js]:
```suggestion
  <%= method.name %>(<%= _.map(method.params, (param) => param.name + ': ' + typeDefs[makeIdForMethodContentDescriptors(method, param)].typeName).join(', ') %>): jayson.JSONRPCResponse {
```

Comment (shanejonas) [reply templates/js/templated/exported-class.template.js]: or
returns a promise that returns the proper type

Comment (shanejonas) [inline templates/js/templated/exported-class.template.js]:
```suggestion
  <%= method.name %>(<%= _.map(method.params, (param) => param.name + ': ' + typeDefs[makeIdForMethodContentDescriptors(method, param)].typeName).join(', ') %>): Promise<<%= typeDefs[makeIdForMethodContentDescriptors(method, method.result)].typeName %>> {
```

File: templates/js/templated/exported-class.template.js [hunk 2]
```diff
@@ -14,21 +19,49 @@ export default class <%= className %> {
 
     this.rpc = jayson.client[options.transport.type](options.transport);
```
Comment (shanejonas) [inline templates/js/templated/exported-class.template.js]:
```suggestion
    this.rpc = jayson.Client[options.transport.type](options.transport);
```

File: templates/js/static/package.json [hunk 3]
```diff
@@ -2,16 +2,23 @@
...
+  ],
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "djv": "^2.1.1",
+    "@open-rpc/schema-utils-js": "git+https://github.com/open-rpc/schema-utils-js.git",
     "jayson": "^2.1.1",
-    "node-fetch": "^2.3.0"
+    "node-fetch": "^2.3.0",
+    "@types/ajv": "^1.0.0",
```
Comment (BelfordZ) [inline templates/js/static/package.json]: ITS FAKE ITS FAKE!!

File: package.json [hunk 4]
```diff
@@ -27,15 +27,16 @@
   "author": "",
   "license": "Apache-2.0",
   "dependencies": {
-    "@open-rpc/schema-utils-js": "^1.0.1",
+    "@open-rpc/schema-utils-js": "git+https://github.com/open-rpc/schema-utils-js.git",
```
Comment (BelfordZ) [inline package.json]:
```suggestion
    "@open-rpc/schema-utils-js": "1.0.2",
```

**Pull request merged**